chalice/deploy/newdeployer.py

```diff
@@ -203,14 +210,17 @@ def deploy(self, config, chalice_stage_name):
         plan = self._plan_stage.execute(resources)
         self._sweeper.execute(plan, config)
         self._executor.execute(plan)
-        return {
-            'stages': {
-                chalice_stage_name: {
-                    'resources': self._executor.resource_values,
-                },
-            },
+        deployed_values = {
+            'resources': self._executor.resource_values,
             'schema_version': '2.0',
+            'backend': self.BACKEND_NAME,
```

Comment:
So I read the overview, but I am still not following why we need the backend parameter. Specifically, how did we use this in the old deployer? Or is this so we can track future deployment methods over time?

Reply:
So we can validate future deployment methods.  I'm thinking specifically for the cfn case.  You can't mix the API based deployer with the cfn deployers so I'd like some way to validate this and warn the user that this isn't supported.

I'd still like the cfn deployer to generate this same file so that things that depend on it (say `chalice url`) don't need to know about specific deployer implementations.

Reply:
Sounds good. Makes sense.